components/ILIAS/Authentication/classes/class.ilAuthLogoutPageEditorSettings.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+class ilAuthLogoutPageEditorSettings
+{
+    private static ?ilAuthLogoutPageEditorSettings $instance = null;
+
+    private array $languages = [];
+
+    private ilSetting $storage;
+
+    private ilLanguage $lng;
+
+    public function __construct()
+    {
+        global $DIC;
+        $this->lng = $DIC->language();
+
+        $this->storage = new ilSetting('logout_editor');
+        $this->read();
+    }
+
+    public static function getInstance(): ilAuthLogoutPageEditorSettings
+    {
+        return self::$instance ?? new ilAuthLogoutPageEditorSettings();
+    }
```

Comment:
```suggestion
    public static function getInstance(): self
    {
        return self::$instance ?? (self::$instance = new self());
    }
```

Reply:
Without any assignment to $instance the null coalescing will never return the anything and always will create a new instance